test/test_primitive_array.cpp

```diff
@@ -54,11 +54,34 @@ namespace sparrow
             using array_test_type = primitive_array<T>;
             array_test_type ar{make_array(values)};
 
+            SUBCASE("empty")
+            {
+                CHECK_FALSE(ar.empty());
+            }
+
             SUBCASE("constructor")
             {
                 CHECK_EQ(ar.size(), 4);
             }
 
+            SUBCASE("at")
```

Comment:
These new tests should be added to all layouts, right ?

Reply:
Yes, but I would prefer to do it in a dedicated PR to avoid making this one too heavy. Notice that since these new functions all rely on `operator[]` that is tested for all layouts, we know that they work for all layouts (for now, I'm not saying that we should not add tests).